code/modules/wiremod/components/list/concat.dm

```diff
@@ -0,0 +1,59 @@
+/**
+ * # Concat List Component
+ *
+ * Concatenates a list with a separator
+ */
+/obj/item/circuit_component/concat_list
+	display_name = "Concatenate List"
+	display_desc = "A component that joins up a list with a separator into a single string."
+
+	/// The input port
+	var/datum/port/input/list_port
+
+	/// The seperator
+	var/datum/port/input/separator
+
+	/// The result from the output
+	var/datum/port/output/output
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+/obj/item/circuit_component/concat_list/Initialize()
+	. = ..()
+	list_port = add_input_port("List", PORT_TYPE_LIST)
+	separator = add_input_port("Seperator", PORT_TYPE_STRING)
+
+	output = add_output_port("Output", PORT_TYPE_STRING)
+
+/obj/item/circuit_component/concat_list/Destroy()
+	list_port = null
+	separator = null
+	output = null
+	return ..()
+
+/obj/item/circuit_component/concat_list/input_received(datum/port/input/port)
+	. = ..()
+	if(.)
+		return
+
+	var/seperator = separator.input_value
+	if(!seperator)
+		return
+
+	var/list/list_input = list_port.input_value
+	if(!list_input)
+		return
+
+	var/string_text = ""
+	var/index = 1
+	for(var/entry in list_input)
```

Comment:
what do you mean?

Reply:
I meant so that everything is converted to a string in the same way, but looks like you intentionally want to hide names from atoms (which the to string component doesn't do).